File: MovieProject/MovieProject.Web/Modules/MovieDB/Movie/MovieColumns.cs
using MovieProject.Modules.MovieDB.Movie;
using MovieTutorial;
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MovieProject.MovieDB.Columns;

[ColumnsScript("MovieDB.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    
    public int MovieId { get; set; }
    [EditLink]
    public string Title { get; set; }
    [Width(300)]

    public string Image { get; set; }
    public string Descrition { get; set; }
    public MovieKind Kind { get; set; }
    //public string GenreName { get; set; }
    [Width(200), GenreListFormatter]
    public List<int> GenreList { get; set; }
}

File: MovieProject/MovieProject.Web/Modules/ServerTypes/MovieDB/MovieColumns.ts
﻿import { GenreListFormatter } from "@/MovieDB/Movie/GenreListFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieKind } from "../Modules/MovieDB.Movie.MovieKind";
import { MovieRow } from "./MovieRow";

export interface MovieColumns {
    MovieId: Column<MovieRow>;
    Title: Column<MovieRow>;
    Image: Column<MovieRow>;
    Descrition: Column<MovieRow>;
    Kind: Column<MovieRow>;
    GenreList: Column<MovieRow>;
}

export class MovieColumns extends ColumnsBase<MovieRow> {
    static readonly columnsKey = 'MovieDB.Movie';
    static readonly Fields = fieldsProxy<MovieColumns>();
}

[MovieKind, GenreListFormatter]; // referenced types